roles/openshift_docker_facts/tasks/main.yml

```diff
@@ -37,3 +40,46 @@
 - set_fact:
     docker_options: "{{ openshift.docker.options | default(omit) }}"
   when: not openshift.docker.hosted_registry_insecure | default(False) | bool
+
+# Avoid docker 1.9 when installing origin < 1.2 or OSE < 3.2 on RHEL/Centos and
+# See: https://bugzilla.redhat.com/show_bug.cgi?id=1304038
+- name: Default to latest docker
+  set_fact:
+    docker_version: ''
+    docker_downgrade: False
+  
+- name: Gather installed version of common package
+  shell: >
+    rpm -q {{ openshift.common.service_type }} --queryformat '%{version}' | cut -d '.' -f1,2 | grep -v 'not installed'
+  register: installed_common_version
+  failed_when: false
+  changed_when: false
+  
+- name: Gather available version of common package
+  shell: >
+    yum list available -e 0 -q "{{ openshift.common.service_type}}" 2>&1 | tail -n +2 | awk '{ print $2 }' | sort -r | tr '\n' ' ' | tail -n 1 | cut -d '.' -f1,2
+  register: available_common_version
+  failed_when: false
+  changed_when: false
```

Comment:
I think this can be replaced with `repoquery --nvr {{ openshift.common.service_type }}` with give you the nvr version of the latest available in configured repos and can be used with version_compare to test against rpm -q